packages/web-components/src/components/ic-chip/ic-chip.tsx

```diff
@@ -139,28 +133,22 @@ export class Chip {
   };
 
   private mouseEnterHandler = (): void => {
-    this.isHovered = true;
+    this.hovered = true;
   };
 
   private mouseLeaveHandler = (): void => {
-    this.isHovered = false;
+    this.hovered = false;
   };
 
   private setChipColour = () => {
-    if (convertToRGBA(this.customColor) !== null) {
-      const colorRGBA = convertToRGBA(this.customColor);
-      const rgbaValue = `rgba(${colorRGBA.r.toString()}, ${colorRGBA.g.toString()}, ${colorRGBA.b.toString()}, ${colorRGBA.a.toString()})`;
-
-      const brightness =
-        (colorRGBA.r * 299 + colorRGBA.g * 587 + colorRGBA.b * 114) / 1000;
-      const foregroundColour =
-        brightness > 133.3505
-          ? "var(--ic-architectural-black)"
-          : "var(--ic-architectural-white)";
-
+    const colorRGBA = convertToRGBA(this.customColor);
+    if (colorRGBA) {
+      const { r, g, b, a } = colorRGBA;
       this.el.setAttribute(
         "style",
-        `--chip-custom-color: ${rgbaValue}; --chip-custom-foreground-color: ${foregroundColour};`
+        `--chip-custom-color: rgba(${r}, ${g}, ${b}, ${a}); --chip-custom-foreground-color: var(--ic-architectural-${
+          (r * 299 + g * 587 + b * 114) / 1000 > 133.3505 ? "black" : "white"
```

Comment:
The rest of this refactor does make things more readable - but the loss of 'brightness' and 'foregroundColour' variables makes this line harder to understand. Would you be happy putting some of those numbers back into variables, so that this setAttribute() function reads nicer?